lib/Exporter/ONNXModelWriter.cpp

```diff
@@ -0,0 +1,1042 @@
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeChannelShuffle(const ChannelShuffleNode *node,
+                                                 GraphType &graph,
+                                                 ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "group", node->getGroup());
+  addValueAttribute(proto, "kernel", node->getKernel());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error
+ONNXModelWriter::writeQuantizationProfile(const QuantizationProfileNode *node,
+                                          GraphType &graph, ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "name", node->getProfiledNodeName());
+  addValueAttribute(proto, "number", node->getProfiledOutputNumber());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeTraceEvent(const TraceEventNode *node,
+                                             GraphType &graph, ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "name", node->getEventName());
+  addValueAttribute(proto, "type", node->getEventType());
+  addValueAttribute(proto, "index", node->getIndex());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeInsertTensor(const InsertTensorNode *node,
+                                               GraphType &graph,
+                                               ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "start", node->getStart());
+  addValueAttribute(proto, "count", node->getCount());
+  addValueAttribute(proto, "axis", node->getAxis());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeChannelwiseQuantizedConvolution(
+    const ChannelwiseQuantizedConvolutionNode *node, GraphType &graph,
+    ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "kernel_shape", node->getKernels());
+  addValueAttribute(proto, "strides", node->getStrides());
+  addValueAttribute(proto, "pads", node->getPads());
+  addValueAttribute(proto, "group", node->getGroup());
+  addValueAttribute(proto, "group_wise", node->getGroupwise() ? 1 : 0);
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeSplat(const SplatNode *node, GraphType &graph,
+                                        ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  addValueAttribute(proto, "value", node->getValue());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeCPUMaxSplat(const CPUMaxSplatNode *node,
```

Comment:
Building with the CPU backend is also optional. Can we wrap the CPU specific nodes in `#ifdef GLOW_WITH_CPU` like we do for OpenCL?